### PR TITLE
Ath 1146 - Implement Rubocop

### PR DESCRIPTION
**Pull request by mckhannah1901**
### JIRA
[Ath1146 - Implement_Rubocop](https://wpengine.atlassian.net/secure/RapidBoard.jspa?rapidView=522&projectKey=ATH&modal=detail&selectedIssue=ATH-1146)

### What was done
- Rubocop is now set up and running 

### How to test
- Run `rubocop -D`
- No offences should occur

### CODE REVIEW

File: landing_page/app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
-      redirect_to login_path 
-    end  
-
-    before_action :set_cache_buster
-    def set_cache_buster
-      response.headers["Cache-Control"] = "no-cache, no-store, max-age=0, must-revalidate"
-      response.headers["Pragma"] = "no-cache"
-      response.headers["Expires"] = "Fri, 01 Jan 1990 00:00:00 GMT"
-    end
+  rescue_from CanCan::AccessDenied do |_exception|
```
Comment (jamesfairbairn-wpengine) [inline landing_page/app/controllers/application_controller.rb]: Currently you're never using this `_exception` variable, so it may be a good idea to see if it can be excluded, or just log the error to the rails console for debugging purposes?

File: landing_page/.rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    - '**/*.rb'
+    - 'archive_posts.rake'
+  Exclude:
+    - bin/**/*
+    - config/developer_configs.rb
+    - config/developer_vars.rb
+    - db/**/*
+    - node_modules/**/*
+    - vendor/**/*
+  TargetRubyVersion: 2.5
```
Comment (alvarocavalcanti) [inline landing_page/.rubocop.yml]: The ruby version specified in the `Gemfile` is `2.6.3`.

File: landing_page/.rubocop.yml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+# Rails:
+#   Enabled: true
```
Comment (alvarocavalcanti) [inline landing_page/.rubocop.yml]: Why did you commented this out? What does this clause, `Rails`, mean?

File: landing_page/.rubocop.yml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    "%w": ()
+
+Style/RaiseArgs:
+  EnforcedStyle: compact
+
+Style/SingleLineMethods:
+  Enabled: false
+
+Style/SymbolArray:
+    Enabled: false
```
Comment (alvarocavalcanti) [inline landing_page/.rubocop.yml]: Why did you choose these?

File: landing_page/.rubocop.yml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+# Temporary Rubocop TODO exclusions:
+# We might want to fix these at some point, but in the mean time we don't want to disable the cop for other files.
```
Comment (alvarocavalcanti) [inline landing_page/.rubocop.yml]: Is this relevant to your code base? Have you assessed the settings below?

File: landing_page/app/helpers/sessions_helper.rb [hunk 6]
```diff
@@ -1,23 +1,20 @@
 module SessionsHelper
+  # Logs the user in
```
Comment (alvarocavalcanti) [inline landing_page/app/helpers/sessions_helper.rb]: Is this comment really necessary? Is the method below readable?

File: landing_page/app/models/ability.rb [hunk 7]
```diff
@@ -16,12 +15,12 @@ def initialize(user)
     #     can :read, :all
     #   end
     #
-    # The first argument to `can` is the action you are giving the user 
+    # The first argument to `can` is the action you are giving the user
```
Comment (alvarocavalcanti) [inline landing_page/app/models/ability.rb]: What do you think of having to fix commented lines as a `rubucop` offense?

Comment (mckhannah1901) [reply landing_page/app/models/ability.rb]: It probably should ignore comments. I'll look into it 👍

File: landing_page/app/helpers/sessions_helper.rb (hunk 6 above)
Comment (mckhannah1901) [reply landing_page/app/helpers/sessions_helper.rb]: The method below is readable so I will remove this line

File: landing_page/.rubocop.yml (hunk 3 above)
Comment (mckhannah1901) [reply landing_page/.rubocop.yml]: I meant to go back and delete this. It's an unrecognised cop in rubocop so that's why I commented it out

File: landing_page/app/helpers/sessions_helper.rb (hunk 6 above)
Comment (mckhannah1901) [reply landing_page/app/helpers/sessions_helper.rb]: Resolved in commit 468cf39e6692825efc60b3b265ef0896ab6f1b4a

File: landing_page/.rubocop.yml (hunk 3 above)
Comment (mckhannah1901) [reply landing_page/.rubocop.yml]: Resolved in commit 468cf39e6692825efc60b3b265ef0896ab6f1b4a

File: landing_page/.rubocop.yml (hunk 2 above)
Comment (mckhannah1901) [reply landing_page/.rubocop.yml]: Resolved in commit 468cf39e6692825efc60b3b265ef0896ab6f1b4a

File: landing_page/app/controllers/application_controller.rb (hunk 1 above)
Comment (mckhannah1901) [reply landing_page/app/controllers/application_controller.rb]: This exception occurs if someone tries to access someone's user profile without being logged in

File: landing_page/.rubocop.yml [hunk 8]
```diff
@@ -0,0 +1,139 @@
+AllCops:
+  Include:
+    - '**/*.rb'
+    - 'archive_posts.rake'
+  Exclude:
+    - bin/**/*
+    - config/developer_configs.rb
+    - config/developer_vars.rb
```
[the remark this answers is not in the document]
Comment (mckhannah1901) [reply landing_page/.rubocop.yml]: They've now been removed

File: landing_page/.rubocop.yml (hunk 4 above)
Comment (mckhannah1901) [reply landing_page/.rubocop.yml]: Only relevant cops are now in this file

File: landing_page/.rubocop.yml (hunk 5 above)
Comment (mckhannah1901) [reply landing_page/.rubocop.yml]: I've now amended this, thanks for the feedback

File: landing_page/app/controllers/application_controller.rb (hunk 1 above)
Comment (jamesfairbairn-wpengine) [reply landing_page/app/controllers/application_controller.rb]: I more meant that the variable itself is never being used, rather than the exception will never occur.

Comment (mckhannah1901) [reply landing_page/app/controllers/application_controller.rb]: Ah I see, I've removed it and it still works so thanks for spotting it 👍

Comment (mckhannah1901) [reply landing_page/app/controllers/application_controller.rb]: commit 00226e8f6aa8d22925c72076234aa5830f0f7cae
